python/res/enkf/jobs/enkf_scaling_get_data.py

```diff
@@ -0,0 +1,173 @@
+    """
+
+    measured_data = _get_data(ert, events.keys)
+
+    measured_data = measured_data.dropna(axis=1)
+
+    measured_data = _filter_on_column_index(measured_data, events.index)
+
+    measured_data = _filter_out_outliers(measured_data, events)
+
+    if measured_data.shape[1] < 2:
+        raise ValueError(
+            "Not enough data points to perform observation scaling"
+            " for keys: {}, need at least 2, have: {}".format(
+                events.keys, measured_data.shape[2]))
+
+    return measured_data
+
+
+def _filter_ensamble_std(ensamble_std, std_cutoff):
+    """
+    Filters on ensamble variation versus a user defined standard
+    deviation cutoff.
+    """
+    return ensamble_std < std_cutoff
+
+
+def _filter_ens_mean_obs(ensamble_mean, ensamble_std, observation_values, observation_std, alpha):
+    """
+    Filters on distance between the observed data and the ensamble mean based on variation and
+    a user defined alpha.
+    """
+    return abs(observation_values - ensamble_mean) > alpha * (ensamble_std + observation_std)
+
+
+def _combine_filters(filters):
+    df = pd.Series()
+    for series in filters:
+        df = series | df
+    return df
+
+
+def _filter_out_outliers(input_data, events):
+    """
+    Goes through the observation keys and filters out outliers. It first extracts
+    key data such as ensamble mean and std, and observation values and std. It creates
+    a filtered index which is a pandas series of indexes where the data will be removed.
+    This can have duplicates of indicies.
+    """
+    output_data = deepcopy(input_data)
+
+    filters = []
+
+    for key in events.keys:
+
+        ens_mean = input_data.loc[key].mean(axis=0)
+        ens_std = input_data.loc[key].std(axis=0)
+        obs_values = input_data.loc["OBS_" + key]
+        obs_std = input_data.loc["STD_" + key]
+
+        filters.append(_filter_ensamble_std(ens_std, events.std_cutoff))
+        filters.append(_filter_ens_mean_obs(ens_mean, ens_std, obs_values, obs_std, events.alpha))
+
+    combined_filter = _combine_filters(filters)
+    return output_data.drop(columns=combined_filter[combined_filter].index)
+
+
+def _filter_on_column_index(df, index_list):
+    """
+    Takes a dataframe and retuns a subset where the columns in index_list are filtered out
+    """
+    df = deepcopy(df)
+    if isinstance(index_list, (list, tuple)):
+        if max(index_list) > df.shape[1]:
+            msg = ("Index list is larger than observation data, please check input, "
+                   "max index list: {} number of data points: {}".format)
+            raise IndexError(msg)
+        return df.iloc[:, list(index_list)]
+    else:
+        return df
+
+
+def _get_general_data(ert, observation_key):
+    ert_obs = ert.getObservations()
+    data_key = ert_obs[observation_key].getDataKey()
+
+    general_data = pd.DataFrame()
+
+    for time_step in range(ert_obs[observation_key].getActiveCount()):
```

Comment:
Consider chaining those `append`s since they are effectively chained as it stands.
```suggestion
    for time_step in range(ert_obs[observation_key].getActiveCount()):
        data = GenDataCollector.loadGenData(ert, "default", data_key, time_step)

        general_data = (
            general_data
            .append(pd.DataFrame(_get_observations(ert_obs, observation_key), index=["OBS_" + observation_key]))
            .append(pd.DataFrame(_get_std(ert_obs, observation_key), index=["STD_" + observation_key]))
            .append(pd.concat([pd.DataFrame([data[key]], index=[observation_key]) for key in data.keys()]))
        )
```

Reply:
:+1: